python/cutlass_cppgen/backend/gemm_operation.py

```diff
@@ -1718,10 +1719,15 @@ def epilogue_schedule_name_3x(self):
     def procedural_name(self):
         """The full procedural name indicates architecture, extended name, tile size, and layout."""
         opcode_class_name = OpcodeClassNames[self.tile_description.math_instruction.opcode_class]
-        if self.api == ApiVersion.v3x and (self.arch >= 90 or self.arch == 11):
-            kernel_name_template = "cutlass{p}_sm{ar}_{op}_{ex}_{tbm}x{tbn}x{tbk}_{cm}x{cn}x{ck}_{l}_{s}_align{al}{k}{e}"
+        if self.api == ApiVersion.v3x and (self.arch >= 90 or is_intel_xe_arch(self.arch)):
+            prefix="sm"
+            if is_intel_xe_arch(self.arch):
+                prefix="Xe"
+            
+            kernel_name_template = "cutlass{p}_{prefix}{ar}_{op}_{ex}_{tbm}x{tbn}x{tbk}_{cm}x{cn}x{ck}_{l}_{s}_align{al}{k}{e}"
             return kernel_name_template.format(
                 p=self.prefix,
+                prefix=prefix,
```

Comment:
nit: you have self.prefix and prefix here referring to different things. Maybe replace (local variable) prefix with arch_prefix?